### PR TITLE
Renovation: make renovation folder

### PR DESCRIPTION
**Pull request by MaximKudriavtsev**
- new folder system
- add tslint

### CODE REVIEW

File: testing/jest/button.tests.tsx [hunk 1]
```diff
@@ -17,7 +17,7 @@ describe('Button', () => {
         const model = new Button();
         model.text = 'My button';
 
-        model.contentRender = ({text}) => (<div className="custom-content">{text+"123"}</div>);
+        model.contentRender = ({text}) => (<div className="custom-content">{text + "123"}</div>);
```
Comment (LazyLahtak) [inline testing/jest/button.tests.tsx]: text + "123" => \`${text}123\` ?

**Review by LazyLahtak — DISMISSED** [review]

File: testing/tests/Renovation/__meta.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "constellation": "viz",
```
Comment (roman-simionov) [inline testing/tests/Renovation/__meta.json:2]: Let's make 'renovation' constellation

File: js/bundles/modules/parts/widgets-base.js [hunk 3]
```diff
@@ -91,4 +89,9 @@ ui.CollectionWidget = require('../../../ui/collection/ui.collection_widget.edit'
 ui.dxDropDownEditor = require('../../../ui/drop_down_editor/ui.drop_down_editor');
 // Reports
 
+// Renovation
+ui.Button = require('../../../renovation/dist/button.j');
```
Comment (roman-simionov) [inline js/bundles/modules/parts/widgets-base.js:93]: This line rewrites classic ui.Button (line 24)

Comment (roman-simionov) [reply js/bundles/modules/parts/widgets-base.js:93]: Let's add `renovation` namespace

**Review by churkin — DISMISSED** [review]

**Review by roman-simionov — CHANGES_REQUESTED** [review]

File: js/renovation/button.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { Component, Prop, React } from "../component_declaration/common";
+import { Component, Prop, React } from '../component_declaration/common';
```
Comment (roman-simionov) [inline js/renovation/button.tsx:1]: This  module can be moved into renovation folder

File: testing/tests/Renovation/__meta.json (hunk 2 above)
Comment (MaximKudriavtsev) [reply testing/tests/Renovation/__meta.json:2]: In the `viz` constellation all tests work faster.

File: js/bundles/modules/parts/widgets-base.js (hunk 3 above)
Comment (MaximKudriavtsev) [reply js/bundles/modules/parts/widgets-base.js:93]: We don't have conflicts here

File: js/renovation/button.tsx (hunk 4 above)
Comment (MaximKudriavtsev) [reply js/renovation/button.tsx:1]: Moved

**Review by LazyLahtak — APPROVED** [review]

**Review by roman-simionov — APPROVED** [review]

**Review by churkin — APPROVED** [review]

**Pull request merged**